be2-scala/src/main/scala/ch/epfl/pop/json/HighLevelProtocol.scala

```diff
@@ -198,6 +199,36 @@ object HighLevelProtocol extends DefaultJsonProtocol {
     }
   }
 
+  implicit object RumorFormat extends RootJsonFormat[Rumor] {
+    final private val PARAM_SENDER_PK: String = "sender_id"
+    final private val PARAM_RUMOR_ID: String = "rumor_id"
+    final private val PARAM_MESSAGES: String = "messages"
+
+    override def read(json: JsValue): Rumor = {
+      val jsonObject: JsObject = json.asJsObject
+      jsonObject.getFields(PARAM_SENDER_PK, PARAM_RUMOR_ID, PARAM_MESSAGES) match {
```

Comment:
maybe here you can do json.asJsObject.getFields directly without the need to create the variable jsonObject